pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go

```diff
@@ -357,6 +357,11 @@ func (m *rangefeedMuxer) establishMuxConnection(
 		return nil, ctx.Err()
 	case <-init.Done():
 		c := init.Get()
+		if c.err != nil {
+			// Remove the mux client from the cache if it
+			// hit an error.
+			m.muxClients.Delete(nodeID)
+		}
```

Comment:
I don't love that we remove by ID here because it isn't clear to me from the flow control that we aren't perhaps removing a newly added _working_ client.  That seems possible even on the existing path that removes the client.

Reply:
My concern at the time was that because the key of the map is a nodeID you don't know if you are removing the same client that just returned you the error.  I've addressed this by moving the deletion to before we set the error, at which point we know that the client in the map is the one that got the error.